test/unit_CartesianProduct.jl

```diff
@@ -198,4 +200,14 @@ for N in [Float64, Float32, Rational{Int}]
     @test absorbing(CartesianProduct) == absorbing(CartesianProductArray) ==
           EmptySet
     @test b × e == e × b == cpa × e == e × cpa == e × e == e
+
+    # intersection
+    set1 = Hyperrectangle(low=[0.;0.], high=[1.; 1.]);
+    set2 = Hyperrectangle(low=[0.5;0.5], high=[1.5; 1.5]);
+
+    ca1 = decompose(set1, blocks=[1, dim(set1)-1])
+
+    intersect1 = intersection(ca1,set2)
```

Comment:
have you timed this, in master and on this branch?

would be nice to add an artificial example of more dimensionality. for example consider a high dim halfspace that is only constrained in 2 variables, and the cartesian product of hyperrectangles in 2-dim blocks.

Reply:
i don't think that this test corresponds to the function of above...